src/parents.js

```diff
@@ -9,14 +10,19 @@ const parentMap = new WeakMap();
  * @param {*} node
  * @param {object} [options]
  * @param {boolean} [options.force] Overwrite existing `parent` properties
+ * @throws {Error} If one of the nodes is not found in its parent
  */
 export function setAll (node, options ) {
 	walk(node, (node, property, parent) => {
-		let ret = set(node, parent, options);
-
-		if (ret === false) {
-			// We assume that if the node already has a parent, its subtree will also have parents
-			return false;
+		try {
+			const ret = set(node, parent, options);
```

Comment:
This reminds me, when we implement passing the params to `set()`, we should also cater to the case where we pass `property` but not `index`.

Reply:
Though `walk()` should probably pass a pointer object. There is no reason why to only pass `property` but not `index`. then we can just use it wholesale 😁

Reply:
Should we implement the changes to `set` in a separate PR?

Reply:
> Though `walk()` should probably pass a pointer object. There is no reason why to only pass `property` but not `index`. then we can just use it wholesale 😁

What do you mean by pointer object?

Reply:
To clarify, when we're talking about adding `property` and `index` to `set` as args, do we want to make them required (i.e. get rid of the property finding code), or use them optionally? I think it makes most sense to make them required; the main use of `set` is in `setAll`, which is able to pass in property and index. Although, perhaps there are enough cases where you'd want to use `set` without knowing `property` or `index`?

Reply:
> > Though `walk()` should probably pass a pointer object. There is no reason why to only pass `property` but not `index`. then we can just use it wholesale 😁
> 
> What do you mean by pointer object?

`{ node, property, index }`

> Should we implement the changes to `set` in a separate PR?

Up to you, I think it's conceivably part of the functionality needed for this one, but it's a weakly held opinion.

> To clarify, when we're talking about adding `property` and `index` to `set` as args, do we want to make them required (i.e. get rid of the property finding code), or use them optionally? I think it makes most sense to make them required; the main use of `set` is in `setAll`, which is able to pass in property and index. Although, perhaps there are enough cases where you'd want to use `set` without knowing `property` or `index`?

If all the current use cases that call `set()` have this information we can make them required for now and implement the finding code as needed, if use cases emerge in the future.

Reply:
Great! I'll make a separate PR for modifying `walk` to pass the pointer object in the callback

Reply:
We're also definitely going to need to make a new function in `children` which returns a list of `{node, property, index}` instead of just a list of nodes